servers/rendering/renderer_rd/forward_mobile/render_forward_mobile.cpp

```diff
@@ -916,6 +916,10 @@ void RenderForwardMobile::_render_scene(RenderDataRD *p_render_data, const Color
 		if (!is_environment(p_render_data->environment) || !environment_get_fog_enabled(p_render_data->environment)) {
 			spec_constant_base_flags |= 1 << SPEC_CONSTANT_DISABLE_FOG;
 		}
+
+		if (!is_environment(p_render_data->environment) || !environment_get_fog_enabled(p_render_data->environment) || environment_get_fog_mode(p_render_data->environment) != RS::EnvironmentFogMode::ENV_FOG_MODE_DEPTH) {
```

Comment:
This needs to be fixed up. I commented what to do above in my comment on the GLES backend